main.go

```diff
@@ -0,0 +1,101 @@
+package main
+
+import (
+	"context"
+	"errors"
+	ginzap "github.com/gin-contrib/zap"
+	"github.com/gin-gonic/gin"
+	"go.uber.org/fx"
+	"go.uber.org/fx/fxevent"
+	"go.uber.org/zap"
+	"gorm.io/driver/sqlite"
+	"gorm.io/gorm"
+	"moul.io/zapgorm2"
+	"net/http"
+	"time"
+)
+
+func provideLogger() *zap.Logger {
+	logger, _ := zap.NewProduction()
+	return logger
+}
+
+func provideRouter(logger *zap.Logger) *gin.Engine {
```

Comment:
Provides the router for the app.

Currently it also includes the glue for the log library